### PR TITLE
Modify struct-markdown generator code to work from different projct roots

### PR DESCRIPTION
**Pull request by SwampDragons**
Modify makefile to call generate code properly, setting project root. 'make generate' now avoids deleting website code generated in the packer plugin sdk.

For now it will be maintainers' responsibility to regenerate this docs code from the packer plugin sdk every release, and commit it to these folders manually.

remove boot command generator code, since this will be handled in the SDK.

### CODE REVIEW

File: cmd/struct-markdown/main.go [hunk 1]
```diff
@@ -27,8 +27,16 @@ func main() {
...
+
+	projectRoot := os.Getenv("PROJECT_ROOT")
+	var paths []string
+	if projectRoot == "" {
+		// fall back to the packer root.
+		paths = strings.SplitAfter(absFilePath, "packer"+string(os.PathSeparator))
+		projectRoot = paths[0]
+	} else {
+		paths = strings.SplitAfter(absFilePath, projectRoot+string(os.PathSeparator))
+	}
```
Comment (azr) [inline cmd/struct-markdown/main.go:39]: Nice !

**Review by azr — APPROVED** [review]
Nice one ! LGTM !

File: Makefile [hunk 2]
```diff
@@ -56,7 +56,6 @@ install-gen-deps: ## Install dependencies for code generation
 	# out code dependencies; so a go mod tidy will remove them again. `go
 	# install` seems to install the last tagged version and we want to install
 	# master.
-	@(cd $(TEMPDIR) && GO111MODULE=on go get github.com/mna/pigeon@master)
```
Comment (azr) [inline Makefile:59]: Do we want to put that one in the sdk folder for now ?

Comment (SwampDragons) [reply Makefile:59]: I've got the new sdk makefile locally in the sdk repo, but I can put it in the sdk folder.

**Pull request merged**